src/generator/AutoRest.Go/Model/MethodGo.cs

```diff
@@ -213,6 +214,10 @@ public string MethodReturnSignature(bool helper)
             return $"{retVal}, {errVal}";
         }
 
+        public bool ShouldValidate() {
+            return !ParameterValidations.IsNullOrEmpty();
```

Comment:
Nit, opening curly brace is in the wrong location.  More importantly shouldn't this return Singleton<GeneratorSettingsGo>.Instance.ClientSideValidation?  Then you can omit the changes in ParameterGo.cs.